### PR TITLE
Application Class (and Education & Experience) and "addapplication" Command [Done, Can merge]

### LINKED ISSUE #95: [FEATURE] add application feature to Eva (backend)

### PR DESCRIPTION
**Pull request by TheSpaceCuber**
Partial implementation of AddApplication command
Classes for Application, Education and Experience are completed.

Syntax should be: addapplication [filepath]

Parses the text file as an application. Link to applicant class not implemented.

Edit 1:
Sorry, will have FileNotFound Exceptions in a lot of places because of the resume parsing requiring to find the file.

### CODE REVIEW

File: src/main/java/com/eva/model/person/applicant/application/Experience.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        this.position = position;
+        this.description = description;
+    }
+
+    private static LocalDate parseDate(String date) throws DateTimeException {
+        int day = Integer.parseInt(date.substring(0, 2));
+        int month = Integer.parseInt(date.substring(2, 4));
+        int year = Integer.parseInt(date.substring(4, 8));
+        return LocalDate.of(year, month, day);
+    }
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/model/person/applicant/application/Experience.java]: perhaps we can abstract this out to dateutil? If the current method doesnt suit what you need you can create another method and another datetimeformatter to suit what you need

Comment (TheSpaceCuber) [reply src/main/java/com/eva/model/person/applicant/application/Experience.java]: yup ok

File: src/main/java/com/eva/logic/parser/AddApplicationCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    String endDate = sc.nextLine().split(" ")[1];
+                    Education edu = new Education(startDate, endDate, schoolName);
+                    eduList.add(edu);
+                }
+            }
+
+            // Experience
+            while (sc.hasNextLine()) {
+                sc.nextLine(); // this should be number
+                String company = sc.nextLine().split(" ")[1];
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/logic/parser/AddApplicationCommandParser.java:64]: What if the individual items exceed one line? Does the code handle that

File: src/main/java/com/eva/model/person/applicant/application/Education.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    /**
+     * Creates an Education object for one institution.
+     * @param startDate The start date of that particular education.
+     * @param endDate The end date of that particular education.
+     * @param schoolName The name of the institution.
+     */
+    public Education(String startDate, String endDate, String schoolName) {
+        requireNonNull(startDate);
+        requireNonNull(endDate);
+        requireNonNull(schoolName);
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/model/person/applicant/application/Education.java]: can use requireAllNonNull method from some common util

File: src/main/java/com/eva/model/person/applicant/application/Experience.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+     * @param startDate The start date.
+     * @param endDate The end date.
+     * @param companyName The name of the company.
+     * @param position The position held.
+     * @param description The description of the job.
+     */
+    public Experience(String startDate, String endDate, String companyName, String position, String description) {
+        requireNonNull(startDate);
+        requireNonNull(endDate);
+        requireNonNull(companyName);
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/model/person/applicant/application/Experience.java]: Same as education

File: src/main/java/com/eva/model/person/applicant/application/Education.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getSchoolName())
+                .append("\nFrom: ")
+                .append(getStartDate())
+                .append(" To: ")
+                .append(getEndDate())
+                .append("\n");
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/model/person/applicant/application/Education.java]: I think for the sake of standardization, we can use the dateToString method in dateUtil to represent dates.

File: src/main/java/com/eva/model/person/applicant/application/Experience.java [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+    @Override
+    public String toString() {
+        final StringBuilder builder = new StringBuilder();
+        builder.append(getCompanyName())
+                .append("\nRole: ")
+                .append(getPosition())
+                .append("\nFrom: ")
+                .append(getStartDate())
+                .append("\nTo: ")
+                .append(getEndDate())
```
Comment (Ben-Hanan) [inline src/main/java/com/eva/model/person/applicant/application/Experience.java]: same as above

**Review by Ben-Hanan — COMMENTED** [review]
Some tweaks required other than that LGTM

File: src/main/java/com/eva/model/person/applicant/application/Education.java (hunk 3 above)
Comment (TheSpaceCuber) [reply src/main/java/com/eva/model/person/applicant/application/Education.java]: ok thanks

File: src/main/java/com/eva/logic/parser/AddApplicationCommandParser.java (hunk 2 above)
Comment (TheSpaceCuber) [reply src/main/java/com/eva/logic/parser/AddApplicationCommandParser.java:64]: Oh right thanks...i will update it to handle job description + catch errors later on.

**Comment by Ben-Hanan** [comment]
LGTM

**Pull request merged**